src/stability_sdk/client.py

```diff
@@ -234,6 +117,197 @@ def open_channel(host: str, api_key: str = None) -> generation_grpc.GenerationSe
+        sampler: generation.DiffusionSampler = generation.SAMPLER_K_LMS,
+        init_image: Optional[np.ndarray] = None,
+        init_strength: float = 0.0,
+        init_noise_scale: float = 1.0,
+        mask: Optional[np.ndarray] = None,
+        masked_area_init: generation.MaskedAreaInit = generation.MASKED_AREA_INIT_ZERO,
+        guidance_preset: generation.GuidancePreset = generation.GUIDANCE_PRESET_NONE,
+        guidance_cuts: int = 0,
+        guidance_strength: float = 0.0,
+    ) -> np.ndarray:
+
+        p = [generation.Prompt(text=prompt, parameters=generation.PromptParameters(weight=weight)) for prompt,weight in zip(prompts, weights)]
+        if init_image is not None:
+            p.append(image_to_prompt(init_image))
+            if mask is not None:
+                p.append(image_to_prompt(mask, is_mask=True))
+
+        step_parameters = {
+            "scaled_step": 0,
+            "sampler": generation.SamplerParameters(cfg_scale=cfg_scale, init_noise_scale=init_noise_scale),
+        }
+
+        begin_schedule = 1.0 if init_image is None else 1.0-init_strength
+        if begin_schedule != 1.0:
+            step_parameters["schedule"] = generation.ScheduleParameters(
+                start=begin_schedule
+            )
+
+        if guidance_preset is not generation.GUIDANCE_PRESET_NONE:
+            guidance_prompt = None
+            guiders = None
+            if guidance_cuts:
+                cutouts = generation.CutoutParameters(count=guidance_cuts)
+            else:
+                cutouts = None
+            step_parameters["guidance"] = generation.GuidanceParameters(
+                guidance_preset=guidance_preset,
+                instances=[
+                    generation.GuidanceInstanceParameters(
+                        guidance_strength=guidance_strength,
+                        models=guiders,
+                        cutouts=cutouts,
+                        prompt=guidance_prompt,
+                    )
+                ],
+            )
+
+        image_params = {
+            "height": height,
+            "width": width,
+            "seed": [seed],
+            "steps": steps,
+            "parameters": [generation.StepParameter(**step_parameters)],
+            "masked_area_init": masked_area_init,
+        }
+        rq = generation.Request(
+            engine_id=self._generate.engine_id,
+            prompt=p,
+            image=generation.ImageParameters(**image_params)
+        )        
+        rq.image.transform.diffusion = sampler
+
+        result = None
+        for resp in self._generate.stub.Generate(rq, wait_for_ready=True):
+            for artifact in resp.artifacts:
+                if artifact.type == generation.ARTIFACT_IMAGE:
+                    nparr = np.frombuffer(artifact.binary, np.uint8)
+                    result = cv2.imdecode(nparr, cv2.IMREAD_COLOR)
+                    break
+
+        # force pixels in unmasked areas not to change
```

Comment:
we should really be ensuring this on the backend in our inpainting implementation. this is probably a fine hack for now, but we shouldn't rely on it. I'm guessing if the mask doesn't blur edges, this will create some ugly edges in the output image, yeah?

Reply:
yep.